x-pack/plugins/apm/public/components/app/dependency_operation_detail_view/index.tsx

```diff
@@ -6,25 +6,141 @@
+import { ResettingHeightRetainer } from '../../shared/height_retainer/resetting_height_container';
+import { push, replace } from '../../shared/links/url_helpers';
+import { SortFunction } from '../../shared/managed_table';
+import { useWaterfallFetcher } from '../transaction_details/use_waterfall_fetcher';
+import { WaterfallWithSummary } from '../transaction_details/waterfall_with_summary';
 import { DependencyOperationDetailTraceList } from './dependency_operation_detail_trace_list';
+import { DependencyOperationDistributionChart } from './dependency_operation_distribution_chart';
+import { maybeRedirectToAvailableSpanSample } from './maybe_redirect_to_available_span_sample';
 
 export function DependencyOperationDetailView() {
   const router = useApmRouter();
 
+  const history = useHistory();
+
   const {
-    query: { spanName, ...query },
+    query,
+    query: {
+      spanName,
+      dependencyName,
+      sampleRangeFrom,
+      sampleRangeTo,
+      kuery,
+      environment,
+      rangeFrom,
+      rangeTo,
+      spanId,
+      waterfallItemId,
+      detailTab,
+      sortField = '@timestamp',
+      sortDirection = 'desc',
+    },
   } = useApmParams('/dependencies/operation');
 
   useDependencyDetailOperationsBreadcrumb();
 
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+
+  const queryWithoutSpanName = omit(query, 'spanName');
+
+  const spanFetch = useFetcher(
+    (callApmApi) => {
+      return callApmApi('GET /internal/apm/dependencies/operations/spans', {
+        params: {
+          query: {
+            dependencyName,
+            spanName,
+            start,
+            end,
+            environment,
+            kuery,
+            sampleRangeFrom,
+            sampleRangeTo,
+          },
+        },
+      });
+    },
+    [
+      dependencyName,
+      spanName,
+      start,
+      end,
+      environment,
+      kuery,
+      sampleRangeFrom,
+      sampleRangeTo,
+    ]
+  );
+
+  const getSortedSamples: SortFunction<DependencySpan> = (
+    items,
+    localSortField,
+    localSortDirection
+  ) => {
+    return orderBy(items, localSortField, localSortDirection);
+  };
+
+  const samples = useMemo(() => {
+    return (
+      getSortedSamples(
+        spanFetch.data?.spans ?? [],
+        sortField,
+        sortDirection
+      ).map((span) => ({
+        spanId: span.spanId,
+        traceId: span.traceId,
+        transactionId: span.transactionId,
+      })) || []
+    );
+  }, [spanFetch.data?.spans, sortField, sortDirection]);
+
+  const selectedSample = useMemo(() => {
+    return samples.find((sample) => sample.spanId === spanId);
+  }, [samples, spanId]);
+
+  const waterfallFetch = useWaterfallFetcher({
+    traceId: selectedSample?.traceId,
+    transactionId: selectedSample?.transactionId,
+    start,
+    end,
+  });
+
+  const queryRef = useRef(query);
+
+  queryRef.current = query;
```

Comment:
Isn't the ref initialised on every render with `query`?

Reply:
... and why do we need it ?

Reply:
It is to get around the dependencies lint rule for the useEffect call. I'd rather do this (which is not great either) than disable the lint rule for the entire call. I don't want to redirect when the page or pageSize changes.